src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -66,10 +66,11 @@ public HubConnection(IConnection connection, IHubProtocol protocol, ILoggerFacto
             _connection.Closed += Shutdown;
         }
 
-        public async Task StartAsync() => await StartAsyncCore().ForceAsync();
+        public async Task StartAsync(CancellationToken cancellationToken = default) => await StartAsyncCore(cancellationToken).ForceAsync();
 
-        private async Task StartAsyncCore()
+        private async Task StartAsyncCore(CancellationToken cancellationToken)
         {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I am actually thinking this is to early. Currently, the idea is that once you start the connection you cannot start it again. With throwing an exception here this principle is actually broken because since if you catch the exception you can start it again...

Reply:
Another way to look at it is that the behavior is not deterministic. If cancellation happens after this line your connection will not be startable but if it happens on this line it will. Removing this line makes it not startable always.